collectors/python.d.plugin/mysql/mysql.chart.py

```diff
@@ -663,6 +734,65 @@ def userstats_chart_template(name):
 DEFAULT_REPL_CHANNEL = ''
 
 
+# Write Set REPlication
+# https://galeracluster.com/library/documentation/galera-status-variables.html
+# https://www.percona.com/doc/percona-xtradb-cluster/LATEST/wsrep-status-index.html
+class WSRepDataConverter:
+    unknown_value = -1
+
+    def convert(self, key, value):
+        if key == 'wsrep_connected':
+            return self.convert_connected(value)
+        elif key == 'wsrep_ready':
+            return self.convert_ready(value)
+        elif key == 'wsrep_cluster_status':
+            return self.convert_cluster_status(value)
+        elif key == 'wsrep_local_state_comment':
+            return self.convert_local_state_comment(value)
+        return value
+
+    def convert_connected(self, value):
+        # https://www.percona.com/doc/percona-xtradb-cluster/LATEST/wsrep-status-index.html#wsrep_connected
+        if value == 'OFF':
+            return 0
+        if value == 'ON':
+            return 1
+        return self.unknown_value
+
+    def convert_ready(self, value):
+        # https://www.percona.com/doc/percona-xtradb-cluster/LATEST/wsrep-status-index.html#wsrep_ready
+        if value == 'OFF':
+            return 0
+        if value == 'ON':
+            return 1
+        return self.unknown_value
+
+    def convert_local_state_comment(self, value):
+        # https://www.percona.com/doc/percona-xtradb-cluster/LATEST/wsrep-status-index.html#wsrep_local_state_comment
+        if value == 'Joining':
+            return 0
+        elif value == 'Donor/Desynced':
+            return 1
+        elif value == 'Joined':
+            return 2
+        elif value == 'Synced':
+            return 3
+        return self.unknown_value
+
+    def convert_cluster_status(self, value):
+        # https://www.percona.com/doc/percona-xtradb-cluster/LATEST/wsrep-status-index.html#wsrep_cluster_status
+        if value == 'Primary':
+            return 0
+        elif value == 'Non-Primary' or value == 'non-Primary':
```

Comment:
The official documentation does not say nothing about `non-Primary`, why are we not testing `non-primary` too?

Reply:
i think it depends on implementation/version/something else

see
https://github.com/netdata/netdata/pull/6962#issuecomment-536513199

Reply:
> Same for Component Status reporting '-1 == unknow' while cluster was in non-Primary state.

Reply:
I understood now, I thought we were missing, because in the link present in the function comments, I could not find the value `non-Primary`.